mri/operators/gradient/base.py

```diff
@@ -0,0 +1,61 @@
+# #############################################################################
+#  pySAP - Copyright (C) CEA, 2017 - 2018                                     #
+#  Distributed under the terms of the CeCILL-B license,                       #
+#  as published by the CEA-CNRS-INRIA. Refer to the LICENSE file or to        #
+#  http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html for details.   #
+# #############################################################################
+
+# Internal Library import
+from .utils import check_lipschitz_cst
+
+# Third party import
+import numpy as np
+from modopt.math.matrix import PowerMethod
+from modopt.opt.gradient import GradBasic
+
+
+class GradBaseMRI(GradBasic):
+    """ Gradient Analysis class for single channel
+        Parameters
+        ----------
+        data: np.ndarray
+            input data array.
+        operator :
+            a function handle that implements M X
```

Comment:
what is `M X`? (same question as for under)

Reply:
I may need help in documenting this, it is matrix and transfer matrix operation in defined function:
||Mx-y||^2
Maybe add the equation on top for reference?

Reply:
transfer matrix ? I think you mean transpose.
But yes definitely say of which function we are taking the gradient.

Also what does function handle mean?

Reply:
Yes transpose. 
Function handle (comes from MATLAB, I am not sure what we call it in python) is basically the name of function.

eg 
```
def my_function_to_do_x
    ....
x = my_function_to_do_x
```
Here both `x` and `my_function_to_do_x` are function handles to the function "my_function_to_do_x"

Reply:
just function then.

Reply:
also here, it only implements `M` (not `M x`).

Reply:
Hmm, I will change this in next PR? all the tests have almost passed, I would hate to have just a minor commit that would restart another set of runs for 30-45 min